include/autoppl/expression/binop.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <type_traits>
+#include <autoppl/util/traits.hpp>
+#include <autoppl/expression/var_expr.hpp>
+#include <autoppl/expression/variable.hpp>
+
+namespace ppl {
+
+template <class LHSValueType, class RHSValueType, class BinaryOp>
```

Comment:
`BinaryOpNode` will basically take in 2 "variable expressions". You can see what the concept is under `util/var_expr_traits.hpp` in my updated branch. Maybe `LHSVarExprType` and `RHSVarExprType` may be more fitting.